common/src/main/java/com/wynntils/features/user/overlays/NpcDialogueOverlayFeature.java

```diff
@@ -165,23 +179,33 @@ private void updateDialogExtractionSettings() {
         }
 
         private void renderDialogue(PoseStack poseStack, String currentDialogue) {
-            List<TextRenderTask> dialogueRenderTask = List.of(new TextRenderTask(currentDialogue, renderSetting));
-
-            float textHeight = FontRenderer.getInstance().calculateRenderHeight(dialogueRenderTask);
-            // Draw a translucent background
-            int colorAlphaRect = 45;
-            RenderUtils.drawRect(
-                    poseStack,
-                    CommonColors.BLACK.withAlpha(colorAlphaRect),
-                    this.getRenderX(),
-                    this.getRenderY(),
-                    0,
-                    this.getWidth(),
-                    textHeight + 10);
+            TextRenderTask dialogueRenderTask = new TextRenderTask(currentDialogue, renderSetting);
+
+            if (stripColors) {
+                dialogueRenderTask.setText(ComponentUtils.stripColorFormatting(dialogueRenderTask.getText()));
+            }
+
+            float textHeight = FontRenderer.getInstance()
+                    .calculateRenderHeight(
+                            dialogueRenderTask.getText(),
+                            dialogueRenderTask.getSetting().maxWidth());
+
+            if (renderBackground) {
```

Comment:
Maybe make the background an alpha value instead of a boolean? Or a color where you can set alpha to completely transparent if you want to disable it.

Reply:
sure